gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/upgrade/initializer/SearchIndexInitializer.java

```diff
@@ -194,15 +199,20 @@ private CompletableFuture<?> runApiIndexationAsync(ExecutorService executorServi
         }
         try {
             if (api.getDefinitionVersion() == DefinitionVersion.V4) {
-                indexable = apiMapper.toEntity(executionContext, api, primaryOwner, null, false);
+                indexable =
+                    apiIndexerDomainService.toIndexableApi(
```

Comment:
What about having the method `toIndexableApi` being able to handler v4 + non-v4 APIs? That way the logic for creating an indexable is in the domain service.

Reply:
I'm not confident enough to handle V2 API 🙈